### PR TITLE
Add support for mixins

### PR DESCRIPTION
**Pull request by TayHobbs**
Adds the ability for Vuese to be used with mixins. Mixins are still not picked up automatically so this functionality is completely optional. Adding **/*.js to the included will allow these to be picked up.

Let me know if you see something you would like changed.

Thanks!

(This PR was made in place of #103, because a lot has changed and to keep git history clean)

### CODE REVIEW

**Comment by TayHobbs** [comment]
@HcySunYang Any thoughts on this PR?

**Comment by HcySunYang** [comment]
@TayHobbs  Thank you for your work, I will review it carefully.

**Comment by TayHobbs** [comment]
@HcySunYang @IWANABETHATGUY I've updated this MR to no longer have conflicts. Please let me know if this is something you would consider including in the library or not. Thank you!

File: packages/parser/__test__/parseJavascript.test.ts [hunk 1]
```diff
@@ -20,11 +20,12 @@ import { Seen } from '../lib/seen'
 
 function getAST(
   fileName: string,
-  babelParserPlugins?: BabelParserPlugins
+  babelParserPlugins?: BabelParserPlugins,
+  jsFile?: boolean
```
Comment (IWANABETHATGUY) [inline packages/parser/__test__/parseJavascript.test.ts]: better to set a default value `false`

**Comment by TayHobbs** [comment]
@IWANABETHATGUY Updated the param

**Comment by IWANABETHATGUY** [comment]
> @IWANABETHATGUY Updated the param
LGTM!

**Comment by IWANABETHATGUY** [comment]
Thank you for this PR

**Comment by BEEugene** [comment]
Hi! How can I add mixins in auto-generation?
Should there be somewhere @vuese tag?

**Comment by BEEugene** [comment]
> Hi! How can I add mixins in auto-generation? Should there be somewhere @vuese tag?

I got it :)
1) Add such pass to configs (it could be a separate file or package.json config) ` "include": ["**/*.vue", "**/mixins/*.js"]`
2) In case there are imports of other mixins and props for your mixins - they will be generated
3) If mixins don't have any props, etc. - you should point the parameters the same way as you do usually with `@vuese`,
e.g. (if you don't - it won't be shown as this shouldn't be parsed by default):
```
...
data() {
    return {
      /**
       * @vuese
       * description
       */
     config: {
...
```

**Comment by BEEugene** [comment]
And does store documentation support from this #103 pull request was also implemented somehow?

**Pull request merged**